torchvision/models/vision_transformer.py

```diff
@@ -158,32 +159,33 @@ def __init__(
         self.num_classes = num_classes
         self.representation_size = representation_size
         self.norm_layer = norm_layer
-        self.conv_stem_layers = conv_stem_layers
 
-        input_channels = 3
-
-        if self.conv_stem_layers is not None:
+        if conv_stem_layers is not None:
             # As per https://arxiv.org/abs/2106.14881
             self.conv_proj = nn.Sequential()
-            prev_channels = input_channels
-            for i, conv_stem_layer in enumerate(self.conv_stem_layers):
-                if isinstance(conv_stem_layer, Dict):
-                    conv_stem_layer_params = ConvStemLayer(**conv_stem_layer)._asdict()
-                elif isinstance(conv_stem_layer, ConvStemLayer):
-                    conv_stem_layer_params = conv_stem_layer._asdict()
+            prev_channels = 3
+            for i, conv_stem_layer in enumerate(conv_stem_layers):
                 self.conv_proj.add_module(
-                    f"conv_{i}",
-                    nn.Conv2d(
+                    f"conv_bn_relu_{i}",
+                    ConvNormActivation(
                         in_channels=prev_channels,
-                        **conv_stem_layer_params,
+                        out_channels=conv_stem_layer.out_channels,
+                        kernel_size=conv_stem_layer.kernel_size,
+                        stride=conv_stem_layer.stride,
+                        padding=conv_stem_layer.padding,
+                        norm_layer=nn.BatchNorm2d,
```

Comment:
I think in the original repo they always use BN, see
https://github.com/facebookresearch/pycls/blob/main/pycls/models/vit.py#L144
https://github.com/facebookresearch/pycls/blob/8c79a8e2adfffa7cae3a88aace28ef45e52aa7e5/pycls/models/blocks.py#L32-L34

Reply:
Yes, I understand that, but this feature is there to allow uses to change the standard BN with SyncBN easily. As-is, the API is limiting because a) the norm_layer provided by the user is not used and b) there is no way to overwrite the BN. I think we need to answer to 2 questions:
1. Are there any scenarios that make it replacing the BN necessary?
2. If it is necessary, it's clear that the default `norm_layer=LayerNorm` can't be used as a drop-in replacement for the norm layers of the ConvStem. Shall we make the layer overwriteable by adding it inside the `ConvStemConfig`?

@fmassa If you have any thoughts on this, please let us know.